mesonbuild/scripts/__init__.py

```diff
@@ -12,10 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: consider switching to pathlib for this
+from pathlib import Path
+import os
+
 def destdir_join(d1: str, d2: str) -> str:
     # c:\destdir + c:\prefix must produce c:\destdir\prefix
-    if len(d1) > 1 and d1[1] == ':' \
-            and len(d2) > 1 and d2[1] == ':':
-        return d1 + d2[2:]
-    return d1 + d2
+    if d1 and os.path.isabs(d2):
+        # TODO: use removeprefix() when depending on Python 3.9
+        d2 = d2[len(Path(d2).anchor):]
+    return os.path.join(d1, d2)
```

Comment:
I've been doing that in other places, seems cleaner? Or do you think it's not always what we want?
```
p = Path(d2)
os.path.join(d1, d2.relative_to(d2.root))
```

Reply:
Dunno if that's better, just like that it's not playing with path as string, but your code is fine too I guess.

Reply:
Ok, moving the discussion in https://github.com/mesonbuild/meson/pull/10180